### PR TITLE
Codespell: config, action + fixing typos it found/fixed

### PR DESCRIPTION
**Pull request by yarikoptic**
On my quest of making the world a better, typo-free, place I dated to look at AFNI. I still think it is tractable and promise to contribute more, if not to bring it over the finish line myself. IMHO what needs to be done:

- [x] gauge interest -- may be typos considered to be a feature and thus not to be disturbed ;)
- [ ] review this diff, adjust config/ignores so we get no false positives
- [ ] fix up ambiguous ones listed below

Then codespell action should turn green and this PR could be merged.
After that - all new typos would be caught, and some enthusiastic person could reduce the list of auto-ignored  words and start fixing words length 5, 4 ... and so on (possibly stopping at 3 or 4 actually since AFNI just likes short variables names too much)

<details>
<summary>Here is the list of remaining ambigous ones which require manual intervention -- join the fun!</summary> 

```shell
./doc/README/README.environment:2866: veriable ==> variable, veritable
./doc/tex/afni_plugins.tex:376: encounted ==> encountered, encounter
./src/3dTsmoothR.c:128: determing ==> determining, determine
./src/afni_plugins.tex:376: encounted ==> encountered, encounter
./src/pcmat.c:65: interm ==> interim, intern
./src/pcmat.c:165: interm ==> interim, intern
./src/pcmat.c:167: interm ==> interim, intern
./src/pcmat.c:168: interm ==> interim, intern
./src/pcmat.c:181: interm ==> interim, intern
./src/pcmat.c:184: interm ==> interim, intern
./src/thd_opentcat.c:168: degress ==> degrees, digress
./src/3dTcat.c:260: resest ==> reset, recessed
./src/3dTcat.c:708: degress ==> degrees, digress
./src/3dTsmooth.c:113: determing ==> determining, determine
./src/3dbucket.c:639: degress ==> degrees, digress
./src/RomanImperator.h:40: Constans ==> Constants, Constant, Constance
./src/RomanImperator.h:53: Constans ==> Constants, Constant, Constance
./src/afni_plugout.c:53: overrided ==> overrode, overridden
./src/afni_plugout.c:73: overrided ==> overrode, overridden
./src/afni_plugout.c:270: happend ==> happened, happens, happen
./src/plug_maskave.c:35: interal ==> internal, interval, integral
./src/3dBrainSync.c:611: firstest ==> firsttest, first test
./src/3dttest++.c:649: provied ==> provide, provided, proved
./src/LiteClue.c:610: poping ==> popping, pooping
./src/LiteClue.c:888: poping ==> popping, pooping
./src/afni.c:3482: hopefull ==> hopeful, hopefully
./src/afni_history_ptaylor.c:4066: bunding ==> binding, bundling, bounding
./src/afni_history_rwcox.c:7944: inbetween ==> between, in between
./src/afni_history_ziad.c:1762: intented ==> intended, indented
./src/fim.c:131: Temporarly ==> Temporarily, Temporary
./src/fim.c:196: Temporarly ==> Temporarily, Temporary
./src/model_conv_PRF_6.c:652: buring ==> burying, burning, burin, during
./src/model_conv_PRF_6_BAD.c:578: buring ==> burying, burning, burin, during
./src/mri_dicom_hdr.c:11148: preceeds ==> precedes, proceeds
./src/mri_nwarp.c:5174: indexs ==> indexes, indices
./src/pbar.c:472: firstest ==> firsttest, first test
./src/plug_4Ddump.c:126: lastest ==> latest, last
./src/plug_delay_V2.c:181: lastest ==> latest, last
./src/plug_extract.c:122: lastest ==> latest, last
./src/plug_realtime.c:1043: throught ==> thought, through, throughout
./src/prf_common_circular.c:820: buring ==> burying, burning, burin, during
./src/readme_env.h:2868: veriable ==> variable, veritable
./src/sonnets.h:899: elipses ==> ellipses, eclipses, ellipsis
./src/sonnets.h:1031: commend ==> comment, command
./src/sonnets.h:1585: wights ==> weights, waits, whites, rights, writes
./src/suma_datasets.c:5944: chaning ==> chaining, changing
./src/thd_correlate.c:814: interection ==> intersection, interaction
./src/thd_getpathprogs.c:190: considere ==> consider, considered
./src/to3d.c:3111: convering ==> converting, covering
./src/vol2surf.c:947: wether ==> weather, whether
./src/3DEdge/src/typedefs.h:29: Differents ==> Different, Difference
./src/R_scripts/MBA.R:927: informatin ==> information, informing
./src/SUMA/SUMA_3dHist.c:624: seeting ==> seating, setting, seething
./src/SUMA/SUMA_3dHist.c:681: seeting ==> seating, setting, seething
./src/SUMA/SUMA_3dSurf2Vol.c:3660: wether ==> weather, whether
./src/SUMA/SUMA_AnalyzeTrace.c:291: pluging ==> plugging, plugin
./src/SUMA/SUMA_Color.c:188: linearily ==> linearly, linearity
./src/SUMA/SUMA_Color.c:3116: vetors ==> vectors, vetoes
./src/SUMA/SUMA_Color.c:3185: alloate ==> allocate, allotted, allot
./src/SUMA/SUMA_Color.c:11659: countours ==> contours, counters
./src/SUMA/SUMA_Color.c:11691: countours ==> contours, counters
./src/SUMA/SUMA_CreateDO.c:6524: desparate ==> desperate, disparate
./src/SUMA/SUMA_CreateDO.c:6935: desparate ==> desperate, disparate
./src/SUMA/SUMA_Load_Surface_Object.c:1311: optimation ==> optimization, optimisation
./src/SUMA/SUMA_Macros.h:601: interects ==> intersects, interacts
./src/SUMA/SUMA_MapIcosahedron.c:98: interations ==> iterations, interactions
./src/SUMA/SUMA_MiscFunc.c:1471: Carefull ==> Careful, Carefully
./src/SUMA/SUMA_ROI2dataset.c:374: throught ==> thought, through, throughout
./src/SUMA/SUMA_clippingPlanes.c:435: dependend ==> dependent, depended
./src/SUMA/SUMA_help.c:782: Groupe ==> Grouped, Group
./src/SUMA/SUMA_suma.c:187: registerd ==> registered, registers
./src/SUMA/SUMA_xColBar.c:11299: multipler ==> multiplier, multiple
./src/SUMA/GLUT/libglut/glut_win.c:36: Tranverse ==> Transverse, Traverse
./src/SUMA/GLUT/libglut/glut_win.c:56: Tranverse ==> Transverse, Traverse
./src/SUMA/GLUT/libglut/glut_event.c:205: Tranverse ==> Transverse, Traverse
./src/avovk/Aclustering.c:518: allready ==> already, all ready
./src/avovk/Aclustering_NOMASK.c:476: allready ==> already, all ready
./src/avovk/Aclustering_floatNOMASK.c:533: allready ==> already, all ready
./src/avovk/@sigseg:275: signle ==> single, signal
./src/avovk/thd_segtools_fNM.c:960: allready ==> already, all ready
./src/gifti/CMakeLists.txt:2: childrens ==> children, children's
./src/matlab/ErrEval.m:12: wether ==> weather, whether
./src/matlab/RemoveExtension.m:12: Trainling ==> Training, Trailing
./src/matlab/GroupAna.m:262: usuall ==> usually, usual
./src/matlab/MakeColorMap.m:6: linearily ==> linearly, linearity
./src/nifti/nifti2/nifti_tool.h:119: chaning ==> chaining, changing
./src/nifti/nifti2/nifti_tool.h:134: chaning ==> chaining, changing
./src/niml/NIML_base.html:988: preceeds ==> precedes, proceeds
./src/poems/poem_rf1.txt:19: wither ==> either, whether, weather
./src/python_scripts/afnipy/abids_lib.py:65: CAREFULL ==> CAREFUL, CAREFULLY
./src/python_scripts/afnipy/afni_util.py:2529: preceeding ==> preceding, proceeding
./src/python_scripts/afnipy/db_mod.py:5840: normall ==> normal, normally
./src/python_scripts/afnipy/db_mod.py:5841: normall ==> normal, normally
./src/python_scripts/afnipy/db_mod.py:6085: normall ==> normal, normally
./src/python_scripts/afnipy/gui_uber_align_test.py:1410: chaning ==> chaining, changing
./src/python_scripts/afnipy/lib_b0_corr.py:280: applie ==> apply, applied
./src/python_scripts/afnipy/lib_surf_clustsim.py:395: indended ==> intended, indented
./src/python_scripts/scripts/gen_ss_review_scripts.py:878: degress ==> degrees, digress
./src/python_scripts/scripts/init_user_dotfiles.py:1040: soures ==> sources, sores, sours, soured
./src/python_scripts/scripts/make_random_timing.py:4082: clases ==> classes, clashes, cases
./src/rickr/serial_helper.c:887: requeset ==> request, requisite
./src/scripts_install/3dLocalUnifize:266: generall ==> generally, general
./src/scripts_install/@animal_warper:5: folling ==> following, falling, rolling
./src/jpeg-6b/structure.doc:796: raison ==> reason, raisin
./src/mpeg_encodedir/bframe.c:643: doesnt ==> doesn't, does not
./src/mpeg_encodedir/specifics.c:567: doesnt ==> doesn't, does not
./src/qhulldir/libqhull.h:637: preceeding ==> preceding, proceeding
./src/qhulldir/poly2.c:3101: determing ==> determining, determine
./src/svm/plug_3dsvm.c:1388: happend ==> happened, happens, happen
./src/XmHTML/lib/common/format.c:2002: wether ==> weather, whether
./src/XmHTML/lib/common/format.c:3581: wether ==> weather, whether
./src/XmHTML/lib/common/images.c:1213: wastefull ==> wasteful, wastefully
./src/XmHTML/lib/common/layout.c:854: managable ==> manageable, manageably
./src/XmHTML/lib/common/paint.c:1308: buttom ==> button, bottom
./src/XmHTML/lib/common/parse.c:557: preceeding ==> preceding, proceeding
./src/XmHTML/lib/common/parse.c:557: preceeding ==> preceding, proceeding
./src/f2c/expr.c:3135: exected ==> executed, expected
./src/f2c/expr.c:3156: exected ==> executed, expected
./tests/scripts/test_guis.py:81: pluging ==> plugging, plugin

```
</details>

### CODE REVIEW

**Comment by mrneont** [comment]
Hi, Yarik-

That's pretty cool, thanks, and indeed I am not a fan of typos either.

I looked to see whether that could be set to check in comments-only, but I couldn't find such functionality.  Is that your understanding of codespell, as well?  I am a bit leery of global changes; I saw it want to change a variable called "acount" to "account", when something like "acount", "bcount", etc. could be perfectly valid.  

It would be nice to think about doing something like this on a smaller section of codebase to start, rather than everything at once.  The current PR fails in the CircleCI, even.

thanks,
pt

**Comment by afni-rickr** [comment]
Indeed, this seems nice.  Agreeing with PT, it would be preferable to do in pieces, and for C files, preferable to restrict to comments.  There is an open request to go after comments in the codespell repo `https://github.com/codespell-project/codespell/issues/1147`, but it does not appear to be addressed.  Parsing a language is probably well beyond the tool.
Given that, maybe we could start with some automated checking on a subset of files (e.g., the afni/doc/README tree), and then apply it to source in tractable chunks.  It might not even be worth trying to write complete rules at this point (e.g. just start with README, and non-CMake *.txt), but to apply it to sets of files more manually.   We have to review it anyway, so doing it in pieces would be reasonable.

**Comment by yarikoptic** [comment]
> when something like "acount", "bcount", etc. could be perfectly valid.

FWIW -- added `acount` to ignores and reran the `codespell -w`. `bcount` so far does not cause any issue since not close to any word (like `account`).

FWIW I was surprised on how few false positives codespell gave me for other projects. Having variable names which are close to real words did cause even some typos which were bugs not picked up.  So overall I would advocate for having codespell checking even the code as well.  Indeed, if decided to proceed -- someone should check/whitelist what causes the test (build passes now) to  fail -- I couldn't quite put finger to it.

**Comment by mrneont** [comment]
Hi, Yarik- Re. false positives, those are really my fear.  I could see using codespell more generally for future CircleCI notifications once we have things well in hand, in order to help with the prevention of spelling issues.  However, to get to that point, I would like to start slowly and verify what is happening along the way.

Should I just start running codespell with your existing exclusion list on a subset of programs, and start verifying things from there?

File: src/R_scripts/TRR.R [hunk 1]
```diff
@@ -604,7 +604,7 @@ outDF <- function(DF, fl) cat(capture.output(DF), file = paste0(fl, '.txt'), sep
 names(lop$dataTable)[names(lop$dataTable)==lop$subject] <- 'subj'
 names(lop$dataTable)[names(lop$dataTable)==lop$Y] <- 'Y'
 names(lop$dataTable)[names(lop$dataTable)==lop$repetition] <- 'rep'
-names(lop$dataTable)[names(lop$dataTable)==lop$condiiton]  <- 'cond'
+names(lop$dataTable)[names(lop$dataTable)==lop$condition]  <- 'cond'
```
Comment (yarikoptic) [inline src/R_scripts/TRR.R:607]: seems to be fixing bug in code

**Comment by yarikoptic** [comment]
> Should I just start running codespell with your existing exclusion list on a subset of programs, and start verifying things from there?

you are the boss!   An alternative good start IMHO could be

- review all commits individually until the very last one where I do "mass fixing" - it should be easy, they could even be merged if no objections . I have send https://github.com/afni/afni/pull/456 for that
- check "manually" what is that test failure about - may be it is just to ensure a typo? I don't have working build/test setup locally to tell
- make an effort to go through large diff in that last commit.  It is quite amazing on how "goodness" `codespell` picks up. I did identify few extra hits worth ignoring but it is not that many and I made through lots of files, even though not through all.

My point is that with just little effort (in matter of hour or two) AFNI could really get rid of LOTS of typos.  the github workflow intention is the same as you want (for circle ci) -- just to alert on new typos/odd variable names.

**Comment by yarikoptic** [comment]
no point to keep this one open. partial one was merged, subsequent fix up PRs also started to come in

**Pull request closed without merging**